client/src/containers/admin/download-materials-data-button/component.tsx

```diff
@@ -29,25 +29,27 @@ const DownloadMaterialsDataButton: React.FC<DownloadMaterialsDataButtonProps> =
     data: originalData,
   }: DownloaderTransformProps): DownloaderTransformProps => {
     const years = uniq(
-      flatten(originalData.map(({ purchases }) => purchases.map(({ year }) => year))).sort(),
+      flatten(originalData?.data?.map(({ purchases }) => purchases.map(({ year }) => year))).sort(),
```

Comment:
This is not wrong and it's fine. But could you also ensure this button is only available when the data is ready?